### PR TITLE
Add link option in PostTitle block

### LINKED ISSUE #23861: FSE: Add links to the post title block when used in the query
With full site editing enabled, the query /loop block renders the blocks that are placed inside it.

Compare
https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/query-loop/index.php#L32

With the latest post block that has links for the titles:
https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/latest-posts/index.php#L95

There needs to be context added so that theme developers can choose when the title 
is a link text for a link leading to the content (archives), and when it is not (single view).

Two things that makes this more difficult is:

- Untitled posts/pages and providing an alternative way to reach the content. Should the "no title" text always be output? Today this is up to the theme author to decide by using filters.
- Theme authors don't always make this title a link even for archives. Sometimes, they design the theme in a way that there is only a read more link.

### PR DESCRIPTION
**Pull request by ntsekouras**
## Description
Resolves: https://github.com/WordPress/gutenberg/issues/23861

Currently `PostTitle` block is rendered as plain text. This PR adds an option to make it a link and also adds attributes for selecting `target` and `rel`.

There is the need to have more `display` options in `FSE` blocks (ex. make featured image a link, manage excerpt length etc..) and this will be really useful for `Query` block.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/e2e-tests/fixtures/blocks/core__post-title.json [hunk 1]
```diff
@@ -4,7 +4,10 @@
         "name": "core/post-title",
         "isValid": true,
         "attributes": {
-            "level": 2
+			"level": 2,
```
Comment (mcsf) [inline packages/e2e-tests/fixtures/blocks/core__post-title.json:7]: Incorrect indentation here.

**Review by mcsf — APPROVED** [review]
I had reviewed and approved this yesterday, but I guess my connection probably failed. :)

File: packages/e2e-tests/fixtures/blocks/core__post-title.json (hunk 1 above)
Comment (StefanXhunga) [reply packages/e2e-tests/fixtures/blocks/core__post-title.json:7]: Thank You for your assistance!

**Review by StefanXhunga — COMMENTED** [review]
Thank You for your help

File: packages/e2e-tests/fixtures/blocks/core__post-title.json (hunk 1 above)
Comment (ntsekouras) [reply packages/e2e-tests/fixtures/blocks/core__post-title.json:7]: Thank you! Fixed.

File: packages/block-library/src/post-title/block.json [hunk 2]
```diff
@@ -12,6 +12,19 @@
 		"level": {
 			"type": "number",
 			"default": 2
+		},
+		"makeLink": {
```
Comment (youknowriad) [inline packages/block-library/src/post-title/block.json:16]: should it be more "isLink"?

Comment (ntsekouras) [reply packages/block-library/src/post-title/block.json:16]: hm.. it seems better yes. Made a follow up here: https://github.com/WordPress/gutenberg/pull/25397

**Pull request merged**